lib/webkit_inspection_protocol.dart

```diff
@@ -250,28 +243,36 @@ class WipResponse {
   String toString() => 'WipResponse $id: $result';
 }
 
-typedef WipEventCallback(WipEvent event);
+typedef WipEvent WipEventTransformer(WipEvent event);
 
 abstract class WipDomain {
-  Map<String, WipEventCallback> _callbacks = {};
+  static final _log = new Logger('WipDomain');
 
-  final WipConnection connection;
-
-  WipDomain(this.connection);
+  Map<String, Stream> _eventStreams = {};
 
-  void _register(String method, WipEventCallback callback) {
-    _callbacks[method] = callback;
+  final WipConnection connection;
+  var _onClosed;
+  Stream<WipDomain> get onClosed => _onClosed;
+
+  WipDomain(WipConnection connection) : this.connection = connection {
+    this._onClosed = new StreamTransformer.fromHandlers(
+        handleData: (event, EventSink sink) {
+      sink.add(this);
+    }).bind(connection.onClose);
   }
 
-  void _handleNotification(WipEvent event) {
-    var f = _callbacks[event.method];
-    if (f != null) f(event);
-  }
+  Stream<WipEvent> eventStream(
```

Comment:
I would actually prefer it to be the equivalent of protected, but since Dart doesn't have that...

Will change to _eventStream. it probably shouldn't be part of the public API at this point.